robot_modeling/DQ_Kinematics.m

```diff
@@ -367,6 +369,29 @@ function set_base_frame(obj, base_frame)
             J = 2*vec4(robot_line - workspace_line)'*Jl;
             
         end
+
+
+        function residual = line_to_line_angle_residual(robot_line, ...
+                 workspace_line, workspace_line_derivative) 
+        % LINE_TO_LINE_ANGLE_RESIDUAL(robot_point, workspace_line,
+        % workspace_line_derivative) returns the residual related to the
+        % moving line orientation in the workspace that is independent of 
+        % the robot motion (i.e., which does not depend on the robot joint
+        % velocities). 
+        %
+        % For more details see Eq. 4.30 of Juan José Quiroz-Omaña."Whole-Body 
+        % Control of Humanoids Robots at Second Order Kinematics Under 
+        % Unilateral Constraints." PhD thesis, Federal University of
+        % Minas Gerais, 2021.
+        % https://repositorio.ufmg.br/handle/1843/38700
+
+                if ~is_line(robot_line) || ~is_line(workspace_line)
+                    error(['robot_line and'...
```

Comment:
We should also test if the line contains only the primary part (i.e., `is quaternion`) because the calculations only require the line direction. Otherwise, if someone provides a line not passing through the origin, the residual calculated below will have a dual part, which is incorrect, but the test will not fail.